torch/autograd/grad_mode.py

```diff
@@ -60,11 +61,16 @@ class no_grad(_DecoratorContextManager):
         >>> z.requires_grad
         False
     """
+    def __init__(self):
+        if not torch._jit_internal.is_scripting():
+            super().__init__()
+        self.prev = False
```

Comment:
we shouldn't be setting `prev = False` for all `_DecoratorContextManager`

Reply:
Isn't this `no_grad.__init__`, not `_DecoratorContextManager.__init__`? I think the point still stands though. Is it better to use `Optional[bool]` here and refine later in `__exit__`?

Reply:
Oh yea, you're right. The way git displays this really looks look its `_DecoratorContextManager` lol. Yea this is fine as is, no need to use Optional.